src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1737,7 +1737,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 
 	@debounce(2000)
 	private async _updateProcessCwd(): Promise<void> {
-		if (this._isDisposed) {
+		if (this._isDisposed || this.shellLaunchConfig.isExtensionOwnedTerminal || this.shellLaunchConfig.isFeatureTerminal) {
```

Comment:
I think the right check here is actually `this.shellLaunchConfig.customPtyImplementation` since we want cwd to be updated on task/debug/extension owned/etc. terminals.

Reply:
It's not getting updated currently for those, for example:

https://github.com/microsoft/vscode/blob/7f2580fa2ee290d1a87bf8ec0789f1bc2b7990f2/src/vs/workbench/contrib/testing/browser/testingOutputTerminalService.ts#L227-L238

or this
https://github.com/microsoft/vscode/blob/6b0442b3152cf4c3144be945bb295163946f4284/src/vs/workbench/api/common/extHostTerminalService.ts#L262-L269

Reply:
`customPtyImplementation` should be set for both of them, One to this:

https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/terminal/browser/terminalProcessExtHostProxy.ts#L12

Other to this:

https://github.com/microsoft/vscode/blob/7f2580fa2ee290d1a87bf8ec0789f1bc2b7990f2/src/vs/workbench/contrib/testing/browser/testingOutputTerminalService.ts#L159

Reply:
Keeping isFeatureTerminal/isExtensionTerminal means no cwd updating which means no fallback mechanism to fetch the cwd if shell integration is disabled for tasks/debug/extension owned terminals.

Reply:
Okay thanks I see. I didn't realize a terminal could simultaneously have `isFeatureTerminal`/`isExtensionOwnedTerminal` and `isCustomPtyImplementation`, but that makes sense